### ISSUE TITLE
Allow differentiation between authors/contributors

### ISSUE THREAD

**Issue by erikvansebille** [citation-file-format/citation-file-format#84]
It seems that CFF (v1.0.3) only implements one type of contributor to the software: an `author`. 

However, it would be useful to be able to differentiate between levels of contribution, a bit like in the [CRediT](https://www.casrai.org/credit.html) system. Is this something to consider for CFF too?

**Comment by jspaaks** [comment]
I'd say it's at least something to consider. Related issue: https://github.com/citation-file-format/citation-file-format/issues/66 . How to do this exactly still needs more discussion. Maybe we could have a ``contributors`` _instead of_ an ``authors``, then create various roles in which people have contributed, one of which could be ``author``, ``funder``, and whatever other roles we think is relevant (also looking at the CRediT roles). Maybe even have a free text field as opposed to an ``enum``.

**Comment by sdruskat** [comment]
Thanks @erikvansebille. This is indeed something that we want to implement in CFF in the future. Have you seen #66? Can I consider this issue a duplicate of #66?

**Comment by sdruskat** [comment]
Thanks @erikvansebille. I think that this will make sense, and indeed I had thought of implementing this in an extension that will make CFF more completely compatible with another metadata format, CodeMeta (in future work).

I'll need to think about whether dropping the `author` field would be viable, as I guess a lot of people are familiar with the (traditional) concept and would be irritated by its absence. There is also the conceptual difference between authors and contributors, as sedimented in author line vs. acknowledgement section in, e.g., a paper, although I don't think that this is something helpful to maintain.

Also, if `zenodo.json` maintains both fields (as mentioned in #66), it may be worth having the two? This also depends on the status of the Zenodo format, about which I know too little. You were mentioning recently, @jspaaks, that this may be nothing sustainable?

**Comment by jspaaks** [comment]
About the supposed sustainability of the Zenodo metadata file format: I don't recall having an opinion on that, although that's not saying much. My memory is less than great, plus some say I have more opinions than sense. But looking at it now, I'd say that the Zenodo metadata format is probably reasonably sustainable. I mean, there's some money behind it, some people are professionally involved with storing and retrieving items based on the metadata, and perhaps most importantly: it is used in practice on a large scale, which AFAIK is not true for either CFF or CodeMeta. 

Some people would argue that it would be better to use a different format ([pick your poison from the crosswalk table over at CodeMeta](https://github.com/codemeta/codemeta/blob/239e644729df4d061bde66fd1cc5e9acde570026/crosswalk.csv)). I'm not sure where I stand on this. Regardless of whether it's better or not, I think it is unrealistic to think that all those formats from the crosswalk will merge within the next 10 years or so.

Last point: (just) the fact that Zenodo has a separate ``author`` and ``contributor`` does not imply that CFF therefore should also have those same keys (or synomyms). With slightly smarter tools you could easily convert to e.g. bibtex. Just a matter of filtering out the contributors that aren't authors, for example with Python code like so:

```python
authors = [contributor for contibutor in contributors if contributor['role']=='author']
```
So technically, this is doable, but it immediately raises the question: should you cite by filtering out anyone who isn't an author (i.e. anyone who isn't a CRediT 13) --I think not. BTW this is what Zenodo does (by default at least). I can see the value of limiting the contributors to a subset of all contributors, but I'd prefer to do this based on significance of the contribution (somewhat arbitrary I know), rather than role (which happens to be the common practice). I see that CRediT has a optional DEGREE OF CONTRIBUTION, somewhat similar to the transitive credits weights that I've seen mentioned but never implemented (probably the FORCE11 docs?)

**Comment by erikvansebille** [comment]
Thanks @sdruskat and @jspaaks for these rapid responses. I guess my Issue indeed overlaps somewhat with #66, although I was hoping for even more flexibility using something like the CRediT system.
But really good to see you're on top of this already and are thinking about implementing. Look forward to see what comes out!

**Comment by jspaaks** [comment]
The upcoming release 1.3.0 of the CFF schema adds a key `contributors` which can help solve the problem where repository owners

1. gift authorship to contributors even when the contribution is insignificant, simply because there is no other mechanism to give thanks OR
2. omit their contributors entirely, thus giving the impression that only they should be credited with the perceived benefits of the software.

However, the current issue as well as some others

1. _Align person roles with the OpenRIF Contribution Role Ontology_ https://github.com/citation-file-format/citation-file-format/issues/27
1. _Create roles as a simple way to record contribution roles?_ https://github.com/citation-file-format/citation-file-format/issues/112

suggest that CFF needs more granularity in differentiating the specific roles that each contributor had. I'm wondering, who or what would benefit from the additional level of detail? How would these metadata be used?

**Comment by jcolomb** [comment]
I think a distinction between authors and contributors (like in zenodo) make sense as one may have done some work (contributed) without feeling responsible for it (for scientific papers, the "authors" officially need to agree for the publication of the work, not the contributors, the corresponding author is supposed to get agreement from everyone before publication and every author is responsible for the publication).

I would like to see roles "attributable" to both authors and contributors (unlike in zenodo) and using some useful taxonomy/taxonomies (unlike in zenodo).

briefly: +1 for author/contributor distinction, AND this is not dependent on the use of roles descriptors.

**Comment by jspaaks** [comment]
Hi Julien, thanks for your quick response. 

> I would like to see roles "attributable" to both authors and contributors (unlike in zenodo) and using some useful taxonomy/taxonomies (unlike in zenodo).

So if I understand this correctly, you would like to be able to write down the specific role that an author or a contributor had. For argument's sake let's say that we add roles to CFF using a taxonomy that perfectly captures the relevant concepts. What I'm then wondering about is, how would that level of detail be used? Are there any systems that could make use of it now / after being updated?

**Comment by jcolomb** [comment]
the short answer is no, I am not aware of any tool using the Credit information (which has been available in xml format for many publishers and for many years), no tool looking at roles indicated in zenodo or data cite data. 
I still have the feeling we should capture the information, even if it not used (yet ?).

PS: these comments would better fit in #112 , as I wrote above, I think this is independent on having the authors/ contributors fields.

**Comment by sdruskat** [comment]
> Hi Julien, thanks for your quick response.
> 
> > I would like to see roles "attributable" to both authors and contributors (unlike in zenodo) and using some useful taxonomy/taxonomies (unlike in zenodo).
> 
> So if I understand this correctly, you would like to be able to write down the specific role that an author or a contributor had. For argument's sake let's say that we add roles to CFF using a taxonomy that perfectly captures the relevant concepts. What I'm then wondering about is, how would that level of detail be used? Are there any systems that could make use of it now / after being updated?

As far as I can tell, InvenioRDM (the platform Zenodo is migrating to in October) has a field "Contributors" with a required _role_ subfield: https://inveniordm.docs.cern.ch/reference/metadata/#contributors-0-n. Docs say "The role of the person or organisation selected from a customizable controlled vocabulary", so probably customizable per instance.

IMHO this provides good incentive to implement roles in CFF, albeit no specific pointers as to which taxonomy to use. Perhaps @lnielsen knows more?

**Comment by sdruskat** [comment]
Locking conversation, please continue discussion in #112.

**Comment by sdruskat** [comment]
Closing as fixed in #439.
